packages/ast/src/clients/rpc/scoped/rpc.ts

```diff
@@ -200,10 +201,32 @@ export const createScopedRpcTmFactory = (
           extensions ? 'client' : 'rpc',
       )
     );
+
     let functionStatements;
+    let awaitClientCreation;
+    
+    if (newClientType) {
```

Comment:
You should really need this case distinction long term. The first case should work everywhere. However, feel free to keep it for a trial-period.

Reply:
I discussed with @Zetazzz and he wants to make it into an option instead of default behavior. wdyt @Zetazzz ?